docs/source/tutorials/cdc_dfu_flash.rst

```diff
@@ -0,0 +1,120 @@
+########################
+USB CDC and DFU Flashing
+########################
+
+Introduction
+------------
+
+Since the ESP32-S2 introduction, Espressif has been working on USB peripheral support for some of the SoC families, including the ESP32-C3 and the ESP32-S3.
+
+This new peripheral allows a lot of new possibilities, including flashing the firmware directly to the SoC without any external USB-to-Serial converter.
+
+In this tutorial, you will be guided on how to use the embedded USB to flash the firmware.
+
+**The current list of supported SoCs:**
+
+========= =======================
+SoC       USB Peripheral Support
+========= =======================
+ESP32-S2  CDC and DFU
+ESP32-C3  CDC only
+ESP32-S3  CDC and DFU
+========= =======================
+
+It's important that your board includes the USB connector attached to the embedded USB from the SoC. If your board doesnt have the USB connector, you can attach an external one to the USB pins.
+
+This instructions will only work on the supported devices with the embedded USB peripheral. The external USB-to-Serial doesn't works on this tutorial.
+
+For a complete reference to the Arduino IDE tools menu, please see the `Tools Menus <../guides/tools_menu.html>`_ reference guide.
+
+USB DFU
+-------
+
+The USB DFU (Device Firmware Upgrade) is a class specification from the USB standard that adds the ability to upgrade the device firmware by the USB interface.
+
+Flashing Using DFU
+******************
+
+.. note::
+    DFU is only supported by the ESP32-S2 and ESP32-S3. See the table of supported SoCs.
+
+To use the USB DFU to flash the device, you will need to configure some settings in the Arduino IDE according to the following steps:
+
+1. Enter into Download Mode manually
+
+This step is done only for the first time you flash the firmware in this mode. To enter into the download mode, you need to press and hold BOOT button and press and release the RESET button.
+
+To check if this procedure was done correctly, now you will see the new USB device listed in the available ports. Select this new device in the **Port** option.
+
+2. Configure the USB DFU
+
+Now you need to set the USB DFU as default on BOOT and for flashing.
```

Comment:
In the next step you can set the USB DFU..

(Sentence "now you need" was used before few times so this change can make the guide more readeble)